### PR TITLE
Improve route guards and error handling (GSI-705)

### PR DESCRIPTION
**Pull request by Cito**
- Checked that all route guards are in place and errors are handled.
- Page refreshes and reloas should work.
- Multiple tabs should work (with MSW, session state is not replicated because this relies on the backend state management).
- Properly handle log in and log out, and improve the testability with MSW.
- Note that we access the cookie only in the MSW code, not in the production code.
- Simplify the fetching of IVAs.
- Update Python to version 3.12 (as we do in the backend).

### CODE REVIEW

**Comment by Cito** [comment]
Ok, will check the "forgot 2FA code" flow.

**Comment by Cito** [comment]
Resetting the authenticator setup should work now.

File: src/mocks/login.js [hunk 1]
```diff
@@ -2,14 +2,14 @@ import { OIDC_AUTHORITY_URL } from "../utils/utils";
 import { user } from "./data";
 
 const SCOPE = process.env.REACT_APP_OIDC_SCOPE;
-const CLIENT_ID = process.env.REACT_APP_OIDC_CLIENT_ID;
+const CLIENT_ID = process.env.REACT_APP// (set this to "NeedsRegistration" to test the full registration and 2FA flow)_OIDC_CLIENT_ID;
```
Comment (ac-jorellanaf) [inline src/mocks/login.js:5]: This seems to have broken the mock login?

**Review by ac-jorellanaf — CHANGES_REQUESTED** [review]
All ok, bar one issue

File: src/mocks/login.js (hunk 1 above)
Comment (Cito) [reply src/mocks/login.js]: Oops, accidentally checked in some garbage that does not belong there. Fixed it now.

**Review by ac-jorellanaf — APPROVED** [review]

**Pull request merged**